src/React.Router/HtmlHelperExtensions.cs

```diff
@@ -123,7 +123,7 @@ var reactComponent
 
 				var executionResult = reactComponent.RenderRouterWithContext(clientOnly, serverOnly);
 
-				if (executionResult.Context?.status != null)
+				if (executionResult.Context?.status != null || executionResult.Context?.url != null)
```

Comment:
Why is this change needed?

Reply:
It is possible for a redirect to be returned [without a status code defined](https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/Redirect.md)